server/app/api/controllers/leaves_application.js

```diff
@@ -41,16 +41,80 @@ module.exports = {
           );
         },
         getAll: function(req, res, next) {
-          Leaves_Application_Model.find({}, function(err, users) {
+          Leaves_Application_Model.find()
+          .populate('employee', 'firstname lastname')
+          .populate('functional_manager', 'firstname lastname')
+          .populate('approved_by', 'firstname lastname')
+          .exec(function(err, users) {
             if (err) {
               next(err);
             } else {
               res.json({
                 status: "success",
-                message: "Users list found!!!",
+                message: "Leave Details found!!!",
                 data: users
               });
             }
           });
+        },
+        
+        getLeaveInfo: function(req, res, next) {   
+       Leaves_Application_Model.findOne({ _id: req.params.id })
+      .populate('employee', 'firstname lastname')
+      .populate('functional_manager', 'firstname lastname')
+      .populate('approved_by', 'firstname lastname')
+      .exec(function(err, leaves) {
+      if (err) {
+        next(err);
+      } else {
+        res.json({
+          status: "success",
+          message: "Employee Leave Information found!!!",
+          data: leaves
+        });
+      }
+    });
+  },
+
+        //Update status to canceled OR approved by field.
+         update: function (req, res, next) {
+        Leaves_Application_Model.findOneAndUpdate({ _id: req.params.id },
+      {
+        $set: req.body,
+        updated_date : new Date(),
+        last_updated_by : req.user.userName,
+      },
+      function (err, info) {
+        if (err) {
+          next(err);
+        }
+        else {
+          res.json({
+            status: "success",
+            message: "Updated Leave Information successfully!!!",
+          });
+        }
+      });
+     },
+
+  delete: function (req, res, next) {
+    Leaves_Application_Model.findOneAndUpdate({ _id: req.params.id },
+      {
+        status: "Inactive",
+        updated_date : new Date(),
+        last_updated_by : req.user.userName
+      },
+      function (err,info) {
```

Comment:
info is not being used

Reply:
Info was assumed as part of syntax as call back function. Removed as it's not being used and api is working fine without info parameter.